### ISSUE TITLE
 snakePipes createEnvs ->  There was an error when creating the environments!

### ISSUE THREAD

**Issue by sunta3iouxos** [maxplanck-ie/snakepipes#812]
Hi all, this haven't happened before. I tried to reinstall the snakepipes into a new server.
After installing and activating the snakePipes environment I get the following error:
```conda create --experimental-solver=libmamba -n snakePipes -c mpi-ie -c bioconda -c conda-forge snakePipes```
```conda activate snakePipes```
Here is the information from ``` snakePipes info```
```
Organism YAML files:
    miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/shared/organisms/SchizoSPombe_ASM294v2.yaml
    miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/shared/organisms/GRCz10.yaml
    miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/shared/organisms/mm9.yaml
    miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/shared/organisms/mm10.yaml
    miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/shared/organisms/hs37d5.yaml
    miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/shared/organisms/dm3.yaml
    miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/shared/organisms/mm10_gencodeM13.yaml
    miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/shared/organisms/hg38.yaml
    miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/shared/organisms/dm6.yaml

The workflow-generic defaults.yaml file is:
    miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/shared/defaults.yaml

The default cluster.yaml file. Its defaults are overridden by the per-workflow cluster.yaml files:
    miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/shared/cluster.yaml

Workflow-specific cluster.yaml and defaults.yaml files are in:
    miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/workflows/createIndices
    miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/workflows/mRNA-seq
    miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/workflows/preprocessing
    miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/workflows/noncoding-RNA-seq
    miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/workflows/HiC
    miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/workflows/DNA-mapping
    miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/workflows/scRNAseq
    miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/workflows/ChIP-seq
    miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/workflows/ATAC-seq
    miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/workflows/WGBS

Temp dir under /scratch/ccg-ngs/tmp/ exists and will be used.
```
When run the snakePipes createEnvs:
```Creating environment from /projects-raptor/ccg-ngs/production/sw/miniconda3/envs/snakePipes/lib/python3.10/site-packages/snakePipes/shared/rules/envs/shared.yaml in /projects-raptor/ccg-ngs/production/sw/miniconda3/envs
There was an error when creating the environments!
```

the context of the shared.yalm are:
```
name: snakepipes_shared_environment_1.0
channels:
 - conda-forge
 - bioconda
dependencies:
 - deeptools = 3.5.0
 - seqtk = 1.3
 - pigz = 2.3.4
 - snpsplit = 0.3.4
 - samtools = 1.10
 - fastqc = 0.11.9
 - cutadapt = 2.8
 - trim-galore = 0.6.5
 - multiqc = 1.10.1
 - fastp = 0.20.0
 - umi_tools = 1.0.1
```
Also, shouldnt some other programs be there? like bwa STAR etc?

**Comment by katsikora** [comment]
Hi,

do you have `mamba` in your base Conda environment?

The shared.yaml lists dependencies that are shared between multiple snakePipes workflows. STAR and other aligners are listed in workflow-specific yaml files and result in building workflow-specific Conda environments on top of the shared one.

Best,
Katarzyna

**Comment by sunta3iouxos** [comment]
Hi Katarzyna,
So I am using --experimental-solver=libmamba as stated [HERE](https://www.anaconda.com/blog/a-faster-conda-for-a-growing-community). This is the conda environment using the mamba solver. I am unsure if this is the issue, but I never experienced a problem before.

Thank you for the second clarification.
Best
Theo

**Comment by sunta3iouxos** [comment]
interesting the following poped-up after:
for both usages of conda, with or without the --experimenta-sorver=libmamba

```
SafetyError: The package for snakepipes located at /projects-raptor/ccg-ngs/production/sw/miniconda3/pkgs/snakepipes-2.5.3-py_0
appears to be corrupted. The path 'site-packages/snakePipes/shared/cluster.yaml'
has an incorrect size.
  reported size: 995 bytes
  actual size: 1003 bytes

SafetyError: The package for snakepipes located at /projects-raptor/ccg-ngs/production/sw/miniconda3/pkgs/snakepipes-2.5.3-py_0
appears to be corrupted. The path 'site-packages/snakePipes/shared/defaults.yaml'
has an incorrect size.
  reported size: 817 bytes
  actual size: 370 bytes
```

**Comment by adRn-s** [comment]
First, keep in mind that `mamba` and `conda --experimental-solver=libmamba` are two different programs. Snakepipes switched to `mamba` after conda was unable to  resolve the built-in YAML files. Around a month after doing so, conda released the 1.0 version of their experimental solver. We still need to adjust to this. For the moment, here are our working installation instructions: https://snakepipes.readthedocs.io/en/latest/content/setting_up.html ... Please try this.

In the meantime, I will try to try installing snakepipes with your method, which is currently unsupported.

**Comment by sunta3iouxos** [comment]
Hi both,
so I used the mamba command and although it through the same error "WARNING Invalid package cache", "SafetyError: The package for snakepipes":
```
 mamba create -n snakePipes -c mpi-ie -c conda-forge -c bioconda snakePipes

WARNING: A conda environment already exists at '/projects-raptor/ccg-ngs/production/sw/miniconda3/envs/snakePipes'
Remove existing environment (y/[n])? y

WARNING Invalid package cache, file '/projects-raptor/ccg-ngs/production/sw/miniconda3/pkgs/snakepipes-2.5.3-py_0/site-packages/snakePipes/shared/cluster.yaml' has incorrect size
WARNING Invalid package cache, file '/projects-raptor/ccg-ngs/production/sw/miniconda3/pkgs/snakepipes-2.5.3-py_0/site-packages/snakePipes/shared/defaults.yaml' has incorrect size

SafetyError: The package for snakepipes located at /projects-raptor/ccg-ngs/production/sw/miniconda3/pkgs/snakepipes-2.5.3-py_0
appears to be corrupted. The path 'site-packages/snakePipes/shared/cluster.yaml'
has an incorrect size.
  reported size: 995 bytes
  actual size: 1003 bytes

SafetyError: The package for snakepipes located at /projects-raptor/ccg-ngs/production/sw/miniconda3/pkgs/snakepipes-2.5.3-py_0
appears to be corrupted. The path 'site-packages/snakePipes/shared/defaults.yaml'
has an incorrect size.
  reported size: 817 bytes
  actual size: 370 bytes
```

the ```snakePipes createEnvs``` command worked as intended.
